privatelink/azure/dns-endpoints.sh

```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+
+#
+# dns-endpoints.sh
+#
+# Output zone records to correctly map to zonal endpoints for Confluent Cloud.
+#
+# Example:
+#
+#   % ./dns-endpoints.sh my-resource-group private-endpoint-1 private-endpoint-2 private-endpoint-3
+#     *                          A    10.0.0.2 10.0.0.3 10.0.0.4
+#     *.az2                      A    10.0.0.2
+#     *.az1                      A    10.0.0.3
+#     *.az3                      A    10.0.0.4
+
+az 1>/dev/null 2>/dev/null
+[[ $? == 127 ]] && echo "warning: please install 'azure-cli'"
+
+if [[ $# < 2 ]]; then
+    echo "usage: $0 <resource-group> <vpc-endpoint(s)..>" 1>&2
+    echo ""
+    echo "example: $0 resource-group private-endpoint-1 private-endpoint-2 private-endpoint-3"
+    exit 1
+fi
+
+resourceGroup=$1
+
+declare -A dnsrecord
+declare allzonerecord
+
+IFS='
+'
+
+for endpoint in "${@:2}"; do
+    nicId=$(az network private-endpoint show \
+    --name $endpoint --resource-group $resourceGroup \
```

Comment:
Just realized, this doesn't probably pass `shellcheck` with pre-commit.  I haven't had time to setup CI on this repo.  Run with `pre-commit` locally and see if `shellcheck` ends up passing.